src/modules/status-list/status-list.service.ts

```diff
@@ -35,53 +46,64 @@ export class StatusListService {
   async addStatusListEntry(
     credential: CredentialDto
   ): Promise<CredentialWithStatusDto> {
-    let statusListEntity = await this.fetchStatusList(credential.id);
+    let statusListEntity = await this.getCredential(credential.id);
 
     if (!statusListEntity) {
-      statusListEntity = StatusListCredential.create({
-        credentialId: credential.id,
-        namespace: credential.credentialSubject.role.namespace,
-      });
-    }
-
-    const statusListCredential = new URL(
-      `${STATUS_LIST_MODULE_PATH}/${statusListEntity.id}`,
-      this.configService.get('STATUS_LIST_DOMAIN')
-    );
+      const statusListCredential = new URL(
+        `${STATUS_LIST_MODULE_PATH}/${credential.id}`,
+        this.configService.get('STATUS_LIST_DOMAIN')
+      ).href;
+
+      const entry = await this.statusListEntryRepository.save(
+        StatusListEntry.create({
+          statusListIndex: '0', // because we are using one-to-one mapping
+          statusListCredential,
+        })
+      );
```

Comment:
To me, I like encapsulating logic in entities. So I would put this in `NamespaceRevocation`. It has the "intelligence" on how to get an entry for a given namespace.
```
     const statusListCredential = new URL(
        `${STATUS_LIST_MODULE_PATH}/${credential.id}`,
        this.configService.get('STATUS_LIST_DOMAIN')
      ).href;
     // https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021entrya
     const entry: StatusList2021EntryDto = {
          type: 'StatusList2021Entry',
          statusPurpose: 'revocation',
          statusListIndex: '0', // because we are using one-to-one mapping
          statusListCredential,
     }
```
I also think that it makes sense to use a domain operation on `credentialWithStatus` as it is the aggregateRoot.
I agree with this [SO post](https://stackoverflow.com/a/1958668) that
> it is a common practice to have your repository only work with the entities that are Aggregate Roots, and this entity is responsible for initializing the other entities.

Even if we need to break aggregate encapsulation a bit to persist the entities, I think it's best to stick to it from a domain standpoint, as this give us more of the "onion architecture"

```suggestion
      const namespace = credential.credentialSubject.role.namespace;
      const namespaceRevocations = await this.namespaceRevocationsRepository.get(namehash(namespace));
      const domain = this.configService.get('STATUS_LIST_DOMAIN');
      const claimedEntry = namespaceRevocation.claimEntry(STATUS_LIST_MODULE_PATH, domain);
      await this.namespaceRevocationsRepository.save(namespaceRevocations);
      
      const credentialWithStatus = new credentialWithStatus(credential.id); // maybe better to do repo.create() here?
      credentialWithStatus.associateEntry(claimedEntry);
      await this.credentialWithStatusRepository.save(credentialWithStatus);
```

Reply:
It all sounds good, but I see couple of drawbacks for our current situation. 
First of all, ssi hub is using different conventions, so it would make code inconsistent. Second problem is we are using here typeorm, so I would first need to find a solution to nicely pass a connection manager to entity class, which is not obvious for me right know and it would take time to figure it out. But I think we don’t have it right now. Probably at the end we will have to make a big refactor for many to many mapping anyway, so maybe then we can think more about it

Reply:
> so I would first need to find a solution to nicely pass a connection manager to entity class
I'm not really understanding why this is necessary. Let's discuss further if you have time